primefaces/src/main/java/org/primefaces/model/file/NativeUploadedFile.java

```diff
@@ -40,7 +38,7 @@ public NativeUploadedFile() {
     }
 
     public NativeUploadedFile(Part source, Long sizeLimit, String webKitRelativePath) {
-        super(source, FileUploadUtils.requireValidFilename(source.getSubmittedFileName()), sizeLimit, webKitRelativePath);
+        super(source, source.getSubmittedFileName(), sizeLimit, webKitRelativePath);
```

Comment:
@melloware I think you introduce this check FileUploadUtils.requireValidFilename but it's already done in AbstractUploadedFile. I fear quality gate might failed because of that, and if so, I guess it'll be a false postive

Reply:
Yep you are right its because Sonar can't tell that later on we actually do properly handle it.